### PR TITLE
The reacher environment on UR5 can now be used with higher speeds and accelerations without incurring protective stops.

### PR DESCRIPTION
**Pull request by homayoonfarrahi**
The UR5 robot expects actuation commands every 8ms. This was violated in the original environment and communicator, causing protective stops. It was fixed by using the stopj command in a few places. In addition, the original environment would change the acceleration value permanently if the robot went out of bounds. It is now reset to the initial value for every actuation.
These improvements were imported from
https://github.com/homayoonfarrahi/cycle-time-study/tree/master/senseact_mod.

### CODE REVIEW

**Comment by csherstan** [comment]
Can you explain how this improves behavior? It's not obvious to me that using the stopj command would result in higher speeds and accelerations without triggering protective stops.

**Comment by homayoonfarrahi** [comment]
By trial and error, I found three scenarios where protective stops were happening in velocity control mode. The first scenario was when a fast-moving UR5 was [suddenly](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-c4976eccd6e91c510a2e84bdcac211ab1589b16f3aa5204354e9a30c8ac88b32L207) [deprived](https://github.com/kindredresearch/SenseAct/blob/master/senseact/devices/ur/ur_communicator.py#L255) of 125Hz speedj commands. This could happen, for instance, if the learner process crashed or the learner's update took longer than 0.5s while the update and env.step(act) were executing synchronously. Or if there was too much delay between the [episode's end](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-eadd2712401433b1ebf5c02dc35382a3d1ca44532369f61e5190139c798d885eL742) and the subsequent reset.

The second scenario was sending [movej commands](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-eadd2712401433b1ebf5c02dc35382a3d1ca44532369f61e5190139c798d885eL358) to a fast-moving arm without stopping it first. The third scenario was sending nan commands resulting from [division-by](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-eadd2712401433b1ebf5c02dc35382a3d1ca44532369f61e5190139c798d885eL567)-[zeros](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-eadd2712401433b1ebf5c02dc35382a3d1ca44532369f61e5190139c798d885eL623). The first two scenarios are fixed by stopping the arm first using the stopj command.

The speed acceleration value was also [changed](https://github.com/kindredresearch/SenseAct/blob/master/senseact/envs/ur/reacher_env.py#L577) [permanently](https://github.com/kindredresearch/SenseAct/blob/master/senseact/envs/ur/reacher_env.py#L633) if UR5 went out of bounds. It is now [reset](https://github.com/kindredresearch/SenseAct/pull/69/files?diff=unified&w=0#diff-eadd2712401433b1ebf5c02dc35382a3d1ca44532369f61e5190139c798d885eR505) to the original value for every actuation.

**Comment by csherstan** [comment]
Ok, sounds good. Will merge.

**Comment by homayoonfarrahi** [comment]
Can you give me write access please? I can't merge without it.

File: senseact/envs/ur/reacher_env.py
Comment (csherstan) [inline senseact/envs/ur/reacher_env.py:606]: I'm not seeing anywhere this variable actually being used. Could we just remove it?

**Review by csherstan — APPROVED** [review]
Looks good, I'm leaving a couple comments of things that would be nice to add to this PR, but aren't necessary.
1. Duplicated lines (559, 562) and (615, 618)
2. Remove `escaped_the_box`?

**Comment by csherstan** [comment]
> Can you give me write access please? I can't merge without it.

Actually, I don't have authority to do that, but I can merge it in myself. However, I had a couple of comments I left in review. Please let me know if you're willing to address them or not. They won't block the merge.

**Comment by homayoonfarrahi** [comment]
Ok, I will address these, do a test run, and let you know.

**Comment by homayoonfarrahi** [comment]
I addressed the comments @csherstan. The test run was good. Please proceed with the merge.

**Pull request merged**
